### PR TITLE
Move add-on registry in its own package

### PR DESCRIPTION
**Pull request by sneridagh**
This builds upon the [typescript-support-in-core](https://github.com/plone/volto/tree/typescript-support-in-core) branch.

It moves the Volto add-on registry to its own package `@plone/registry`. It will allow to have the feature portable to other frameworks other than Volto itself. Internally, Volto uses it by path, so it does not relies on a released version.
It uses TypeScript, and an initial typing of the registry is in place already.

It also features the same treatment for the `@plone/scripts` use of `i18n` script.

### CODE REVIEW

**Review by davisagli — COMMENTED** [review]
@sneridagh What is the benefit of moving it? Won't it potentially cause issues with hoisting?

**Comment by sneridagh** [comment]
@davisagli World domination! ;) This will allow us to use the add-on registry somewhere else outside Volto. I'm experimenting with adding it to a NextJS build. I have a PoC already working :) It's quite promising.

File: __tests__/webpack-relative-resolver.test.js [hunk 1]
```diff
@@ -1,5 +1,5 @@
 const path = require('path');
-const AddonConfigurationRegistry = require('../addon-registry');
+const AddonConfigurationRegistry = require('../packages/registry/addon-registry');
```
Comment (tiberiuichim) [inline __tests__/webpack-relative-resolver.test.js:2]: this should not depend on this current repository layout, rather use the package name, if possible.

**Comment by mtoepfl** [comment]
I`m going to test this branch with my mtoepfl-typescript branch. 

In which I try to "merge" my work from last year sprint in Namur, where I type some basics like src/helpers/index.ts and first blocks like Title and Listing. 

I put the types in extra .d.ts files where possible, so the "confusion" for none-typescript developers is minimal. Also if they shadow core-blocks and they are able to inherit from this types easily and extend it.

I`m looking forward for your feedback and to discuss this changes in person soon!

File: packages/registry/src/types/index.ts [hunk 2]
```diff
@@ -1,14 +1,26 @@
...
+export type SlotsConfig = {
   [key: string]: unknown;
 };
-export type ViewsConfig = {};
-export type WidgetsConfig = {};
-export type AddonReducersConfig = {};
-export type AddonRoutesConfig = {};
-export type SlotsConfig = {};
+
 export type ComponentsConfig = {
```
Comment (mtoepfl) [inline packages/registry/src/types/index.ts:20]: FYI you can write this more readable as: 

`export type ComponentsConfig = Record<string, { component: React.ComponentType }>;`

As far as I know, the used syntax is prefered if you extend this types with other attributes, eg.:

```
export type ComponentsConfig = {
   [key: string]: { component: React.ComponentType };
   name2: { component: Function; };
   name3: { component: string; };
};
```

File: packages/registry/src/types/Blocks.ts [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+   */
+  edit: React.ComponentType;
+  /**
+   * The group (blocks can be grouped, displayed in the chooser)
+   */
+  schema?: object;
+  /**
+   * The group of the block
+   */
+  blockSchema?: ({ props, intl }: { props: unknown; intl: unknown }) => object;
```
Comment (mtoepfl) [inline packages/registry/src/types/Blocks.ts:42]: Another readability thingy, not needed twice:

`blockSchema?: ({ props, intl }: { props: unknown; intl: unknown }) => object;`

`blockSchema?: (payload: { props: unknown; intl: unknown }) => object;`

also `Record<string, unkown>` instead of `object` seams more explicit to me in this cases.

File: packages/registry/src/index.ts [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+  constructor() {
+    if (!Config.instance) {
+      this._data = {};
+      Config.instance = this;
+    }
+
+    return Config.instance;
+  }
+
+  set(registry: keyof ConfigData, item: ConfigData[keyof ConfigData]) {
```
Comment (mtoepfl) [inline packages/registry/src/index.ts:29]: I`m not sure, I`m only on github web-view, but try:

`set<RegistryKey extends keyof ConfigData>(registry: RegistryKey, item: ConfigData[RegistryKey]) {`

**Review by mtoepfl — COMMENTED** [review]
I did a first review. Sorry for being nitpicking, but as I understood you asked me for feedback especially on typings.

As said, I will test it out. But in my opinion this type of utility can show the benifits of strong typing very well.

File: packages/registry/src/index.ts (hunk 4 above)
Comment (sneridagh) [reply packages/registry/src/index.ts:29]: It worked! What's the rationale behind?

Comment (mtoepfl) [reply packages/registry/src/index.ts]: @sneridagh Nice! This ist a Generic. Something I miss in mypy in this extent. Sorry I'm not really good in explaining things in an easy way, but TypeScript docs are really good.

https://www.typescriptlang.org/docs/handbook/2/generics.html

Please don't hesitate to ping me, if you face an other TypeScript problem. As you did with the comment or also "as any" ist your friend. I do this very frequently and come back later.

**Comment by sneridagh** [comment]
@mtoepfl I added your suggestions.

I also updated the latest updates from `@plone/client` in here. I'm foreseeing a new package `@plone/types` because we will need to share the types across packages outside Volto. For now it's easier to develop types where the action is happening (which I think is the client). We can postpone it until they are more "stable" for now it's fine to sync them from time to time.

@stevepiercy I've added a README here, based on other docs that we have and I made it "agnostic" not tied to Volto necessarily. Please take a look and sorry in advance, I did it quickly and might have lots of inconsistencies. I might have another look tomorrow. I would like to have this excerpt in here because I want the package to be self-explaining (eg. someone that comes out of the blue and must understand the capabilities of this package.

File: news/4949.feature [hunk 5]
```diff
@@ -0,0 +1 @@
+Move add-on registry in its own package @sneridagh
```
Comment (stevepiercy) [inline news/4949.feature:1]:
```suggestion
Moved add-on registry to its own package. @sneridagh
```

File: news/4662.feature [hunk 6]
```diff
@@ -0,0 +1 @@
+TypeScript support in core @sneridagh @ninanoleto
```
Comment (stevepiercy) [inline news/4662.feature:1]:
```suggestion
Added TypeScript support in core. @sneridagh @ninanoleto
```

File: packages/registry/create-theme-addons-loader.js [hunk 7]
```diff
@@ -1,4 +1,4 @@
-const path = require('path');
+// const path = require('path');
```
Comment (stevepiercy) [inline packages/registry/create-theme-addons-loader.js:1]: Why not remove instead of comment?

```suggestion
```

File: packages/registry/package.json [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+  ],
+  "license": "MIT",
+  "version": "1.0.0",
+  "repository": {
+    "type": "git",
+    "url": "git@github.com:plone/volto.git"
+  },
+  "bugs": {
+    "url": "https://github.com/plone/volto/issues",
+    "email": "plone-developers@lists.sourceforge.net"
```
Comment (stevepiercy) [inline packages/registry/package.json:19]: Also remove previous comma to avoid syntax error.

```suggestion
```

File: packages/registry/README.md [hunk 9]
```diff
@@ -0,0 +1,190 @@
+# Add-on and configuration registry
+
+This package provides support for building an add-on registry and infrastructure for JS/TS-based apps.
```
Comment (stevepiercy) [inline packages/registry/README.md:3]:
```suggestion
This package provides support for building an add-on registry and infrastructure for JavaScript and TypeScript-based apps.
```

File: packages/registry/README.md [hunk 10]
```diff
@@ -0,0 +1,190 @@
...
+
+## Extensibility and puggability
+
+As developers, when you build an app, regardless of the framework and technologies used, it's a one-off app. Meaning, you have to build something that has very specific requirements, behavior and look and feel.
```
Comment (stevepiercy) [inline packages/registry/README.md:7]:
```suggestion
As a developer when you build an app, regardless of the framework and technologies used, it's a one-off app.
That means you have to build something that has very specific requirements, behavior, and look and feel.
```

File: packages/registry/README.md [hunk 11]
```diff
@@ -0,0 +1,190 @@
...
+
+Other few times, you need to build something generic, that could be pluggable and extensible. In the JS/TS ecosystem this is often quite complex and the existing frameworks do not provide the meanings to "build up" in these regards.
```
Comment (stevepiercy) [inline packages/registry/README.md:9]:
```suggestion
Sometimes you need to build something generic that is pluggable and extensible.
In the JavaScript and TypeScript ecosystem, this is often quite complex, and the existing frameworks do not provide the means to do this.
```

File: packages/registry/README.md [hunk 12]
```diff
@@ -0,0 +1,190 @@
...
+
+## Add-on registry
+
+An add-on registry is a facility that allows an app built on an existing framework to build on extensibility and pluggability.
```
Comment (stevepiercy) [inline packages/registry/README.md:13]:
```suggestion
An add-on registry is a facility that allows an app that was built on an existing framework to be extensible and pluggable.
```

File: packages/registry/README.md [hunk 13]
```diff
@@ -0,0 +1,190 @@
...
+
+The add-on registry is a store where you can register a number of add-ons that your app is going to consume.
```
Comment (stevepiercy) [inline packages/registry/README.md:15]:
```suggestion
The add-on registry is a store where you can register a number of add-ons that your app consumes.
```

File: packages/registry/README.md [hunk 14]
```diff
@@ -0,0 +1,190 @@
...
+
+Add-on packages are just CommonJS packages. The only requirement is that
```
Comment (stevepiercy) [inline packages/registry/README.md:17]:
```suggestion
Add-on packages are just CommonJS packages.
```

File: packages/registry/README.md [hunk 15]
```diff
@@ -0,0 +1,190 @@
...
+they point the `main` key of their `package.json` to a module that exports, as a default function that acts as a configuration loader.
```
Comment (stevepiercy) [inline packages/registry/README.md:18]:
```suggestion
The only requirement is that they point the `main` key of their `package.json` to a module that exports as a default function, which acts as a configuration loader.
```

File: packages/registry/README.md [hunk 16]
```diff
@@ -0,0 +1,190 @@
...
+
+An add-on can be published in an npm registry, just as any other package. However, add-ons are meant to not be transpiled. They should be released as "source" packages.
```
Comment (stevepiercy) [inline packages/registry/README.md:20]:
```suggestion
An add-on can be published in an npm registry, just as any other package.
However, add-ons are meant to not be transpiled.
They should be released as source packages.
```

File: packages/registry/README.md [hunk 17]
```diff
@@ -0,0 +1,190 @@
...
+
+## Register an add-on
+
+You should declare in your project that you are using an add-on.
```
Comment (stevepiercy) [inline packages/registry/README.md:24]:
```suggestion
You should declare your add-on in your project.
```

File: packages/registry/README.md [hunk 18]
```diff
@@ -0,0 +1,190 @@
...
+This is done in the åpp's `package.json`, `addons` key:
```
Comment (stevepiercy) [inline packages/registry/README.md:25]:
```suggestion
This is done in your app's `package.json`'s `addons` key:
```

File: packages/registry/README.md [hunk 19]
```diff
@@ -0,0 +1,190 @@
...
+  "name": "my-app-project",
+  "addons": [
+    "acme-volto-foo-addon",
+    "@plone/some-addon",
+    "collective-another-volto-addon"
+  ]
+}
+```
+
+By including the add-on name in the `addons` key, the add-on's main default export function is executed, being passed the configuration registry. In that function, the add-on can customize the registry. The function needs to return the `config` (configuration registry) object, so that it's passed further along to the other add-ons.
```
Comment (stevepiercy) [inline packages/registry/README.md:38]:
```suggestion
The `addons` key ensures the add-on's main default export function is executed, being passed the configuration registry.
In that function, the add-on can customize the registry.
The function needs to return the `config` (configuration registry) object, so that it's passed further along to the other add-ons.
```

File: packages/registry/README.md [hunk 20]
```diff
@@ -0,0 +1,190 @@
...
+
+The add-ons are registered in the order they are found in the `addons` key. The last add-on takes precedence over the others. This means that if you configure something in `acme-volto-foo-addon` and then the same in `collective-another-volto-addon`, the latter configured will win and that configuration applied.
```
Comment (stevepiercy) [inline packages/registry/README.md:40]:
```suggestion
The add-ons are registered in the order they are found in the `addons` key.
The last add-on takes precedence over the others.
This means that if you configure something in `acme-volto-foo-addon`, then the same thing later in `collective-another-volto-addon`, the latter configured thing will win and its configuration will be applied.
```

File: packages/registry/README.md [hunk 21]
```diff
@@ -0,0 +1,190 @@
...
+
+The default export of any add-on main `index.js` file should be a function with the signature `config => config`. That is, it should take the configuration registry object and return it, possibly mutated or changed.
```
Comment (stevepiercy) [inline packages/registry/README.md:42]:
```suggestion
The default export of any add-on main `index.js` file should be a function with the signature `config => config`.
That is, it should take the configuration registry object and return it, possibly mutated or changed.
```

File: packages/registry/README.md [hunk 22]
```diff
@@ -0,0 +1,190 @@
...
+
+## Configuration registry
+
+The add-on registry is complemented with the configuration registry.
```
Comment (stevepiercy) [inline packages/registry/README.md:46]: I think you mean supplement, not complement. They are subtly different, where supplement is more like "in addition to", whereas complement is more like "makes complete".

```suggestion
The configuration registry supplements the add-on registry.
```

File: packages/registry/README.md [hunk 23]
```diff
@@ -0,0 +1,190 @@
...
+
+It is a facility that stores app configuration to be shared in the app. This is where the extensibility and pluggability story for our app lies.
```
Comment (stevepiercy) [inline packages/registry/README.md:48]: "story"? maybe "store"?

```suggestion
It is a facility that stores app configuration to be shared in the app.
This is where the extensibility and pluggability store for our app lives.
```

File: packages/registry/README.md [hunk 24]
```diff
@@ -0,0 +1,190 @@
...
+
+Let's say that your app is the UI of a content management system (CMS). This CMS uses blocks as its main fundamental unit of content. The pages that the CMS builds are made up of these blocks. The CMS has some basic available blocks, but it's a requirement that integrators are able to register more blocks in a pluggable way.
```
Comment (stevepiercy) [inline packages/registry/README.md:50]:
```suggestion
Let's say that your app is the user interface of a content management system (CMS).
This CMS uses blocks as its main fundamental unit of content.
The pages that the CMS builds are made up of these blocks.
The CMS has some basic available blocks, but it's a requirement that integrators are able to register more blocks in a pluggable way.
```

File: packages/registry/README.md [hunk 25]
```diff
@@ -0,0 +1,190 @@
...
+
+This app will use the add-on registry to extend the basic CMS capabilities, so an external add-on will be able to sum up with their own add-on(s) to the basic CMS ones.
```
Comment (stevepiercy) [inline packages/registry/README.md:52]:
```suggestion
This app will use the add-on registry to extend the basic CMS capabilities, so an external add-on can supplement their own add-on(s) to the basic CMS ones.
```

File: packages/registry/README.md [hunk 26]
```diff
@@ -0,0 +1,190 @@
...
+
+Let's assume we've defined a key in the registry `config.blocks.blocksConfig` and defined a way to register the available blocks in the CMS as the keys in that object in the configuration registry:
```
Comment (stevepiercy) [inline packages/registry/README.md:54]:
```suggestion
Let's assume we've defined a key in the registry `config.blocks.blocksConfig`, and defined a way to register the available blocks in the CMS as the keys in that object in the configuration registry:
```

File: packages/registry/README.md [hunk 27]
```diff
@@ -0,0 +1,190 @@
...
+    view: FAQBlockView,
+    icon: chartIcon,
+    group: 'common',
+    restricted: false,
+    mostUsed: true,
+    sidebarTab: 1,
+  };
+  ```
+
+The configuration registry will have other keys already set by default which will compose the initial set of basic blocks used by the CMS. The CMS will populate the available blocks in the CMSUI properly to be used by the users.
```
Comment (stevepiercy) [inline packages/registry/README.md:70]:
```suggestion
The configuration registry will have other keys already set by default, which will compose the initial set of basic blocks used by the CMS.
Then the CMS will properly populate the available blocks in the user interface.
```

File: packages/registry/README.md [hunk 28]
```diff
@@ -0,0 +1,190 @@
...
+
+The add-on is meant to extend the initial configuration, so from the default export function of our add-on we should provide the configuration of the new block:
```
Comment (stevepiercy) [inline packages/registry/README.md:72]:
```suggestion
The add-on is meant to extend the initial configuration.
From the default export function of our add-on, we should provide the configuration of the new block:
```

File: packages/registry/README.md [hunk 29]
```diff
@@ -0,0 +1,190 @@
...
+    restricted: false,
+    mostUsed: true,
+    sidebarTab: 1,
+  };
+
+  return config;
+}
+```
+
+Once the app starts, the add-on registry will execute in order all the registered add-ons default export functions, configuring the new block and make it available to the CMS when it asks to the configuration registry for it.
```
Comment (stevepiercy) [inline packages/registry/README.md:92]: Too many pronouns make it difficult to understand what "it" refers to. Here is my guess.

```suggestion
Once the app starts, the add-on registry will execute, in order, all the registered add-ons' default export functions, configuring the new block, and the add-on available to the CMS when it asks the configuration registry for it.
```

File: packages/registry/README.md [hunk 30]
```diff
@@ -0,0 +1,190 @@
...
+```ts
+import config from '@plone/registry'
+
+const blocksConfig = config.blocks.blocksConfig
+```
+
+## Component registry
+
+The configuration registry also stores a components registry in itself.
+The components registry is a mapping of name to component, so you can lookup
```
Comment (stevepiercy) [inline packages/registry/README.md:107]:
```suggestion
The components registry is a mapping of name to component.
```

File: packages/registry/README.md [hunk 31]
```diff
@@ -0,0 +1,190 @@
...
+a name and receive a component that you can reference in your code.
```
Comment (stevepiercy) [inline packages/registry/README.md:108]:
```suggestion
You can look up a name, and receive a component that you can reference in your code.
```

File: packages/registry/README.md [hunk 32]
```diff
@@ -0,0 +1,190 @@
...
+The idea behind is to provide an alternative and more convenient way to customize components.
```
Comment (stevepiercy) [inline packages/registry/README.md:109]:
```suggestion
This provides an alternative, and more convenient, way to customize components.
```

File: packages/registry/README.md [hunk 33]
```diff
@@ -0,0 +1,190 @@
...
+You can override programatically such registrations from your add-on or projects because it's stored in the configuration registry.
+You can customize a component without using shadowing at all, if the code that uses the component retrieves from the component registry, rather then import it directly.
+You can even have modifiers to the component registrations: dependencies. So you can "adapt" the call, given an array of such dependencies.
```
Comment (stevepiercy) [inline packages/registry/README.md:112]:
```suggestion
You can even have modifiers to the component registrations through dependencies.
Thus you can adapt the call, given an array of such dependencies.
```

File: packages/registry/README.md [hunk 34]
```diff
@@ -0,0 +1,190 @@
...
+
+## Registering components by name using `config.registerComponent`
```
Comment (stevepiercy) [inline packages/registry/README.md:114]:
```suggestion
## Register components by name using `config.registerComponent`
```

File: packages/registry/README.md [hunk 35]
```diff
@@ -0,0 +1,190 @@
...
+
+Typically from an add-on or project configuration:
```
Comment (stevepiercy) [inline packages/registry/README.md:116]:
```suggestion
You can register components by name, typically from an add-on or project configuration:
```

File: packages/registry/README.md [hunk 36]
```diff
@@ -0,0 +1,190 @@
...
+```js
+import MyToolbarComponent from './MyToolbarComponent'
+
+config.registerComponent({
+  name: 'Toolbar',
+  component: MyToolbarComponent,
+});
+```
+
+## Retrieving a component from the component registry
```
Comment (stevepiercy) [inline packages/registry/README.md:127]:
```suggestion
## Retrieve a component from the component registry
```

File: packages/registry/README.md [hunk 37]
```diff
@@ -0,0 +1,190 @@
...
+
+You can programatically retrieve a component from the registry using `config.getComponent`:
+
+```js
+const Toolbar = config.getComponent('Toolbar').component
+```
+
+or by using the convenience component `Component` if you want to use it in JSX directly
```
Comment (stevepiercy) [inline packages/registry/README.md:135]:
```suggestion
Or you can retrieve a component by using the convenience component `Component`, if you want to use it in JSX code directly.
```

File: packages/registry/README.md [hunk 38]
```diff
@@ -0,0 +1,190 @@
...
+
+```jsx
+import Component from '@plone/volto/components/theme/Component/Component';
+
+<Component componentName="Toolbar" {...props} />
+```
+
+Please notice that you are able to pass `props` down to the retrieved component.
```
Comment (stevepiercy) [inline packages/registry/README.md:143]:
```suggestion
Note that you can pass `props` down to the retrieved component.
```

File: packages/registry/README.md [hunk 39]
```diff
@@ -0,0 +1,190 @@
...
+
+## Adapting the component using `dependencies` array
```
Comment (stevepiercy) [inline packages/registry/README.md:145]:
```suggestion
## Adapt the component using the `dependencies` array
```

File: packages/registry/README.md [hunk 40]
```diff
@@ -0,0 +1,190 @@
...
+
+You can register components, and retrieve them afterwards given a list of modifiers `dependencies`.
```
Comment (stevepiercy) [inline packages/registry/README.md:147]:
```suggestion
You can register components, then retrieve them, given a list of modifiers using `dependencies`.
```

File: packages/registry/README.md [hunk 41]
```diff
@@ -0,0 +1,190 @@
...
+import MyTeaserNewsItemComponent from './MyTeaserNewsItemComponent'
+
+config.registerComponent({
+    name: 'Teaser',
+    component: MyTeaserNewsItemComponent,
+    dependencies: 'News Item',
+  });
+```
+
+and then retrieve it:
```
Comment (stevepiercy) [inline packages/registry/README.md:159]:
```suggestion
And then retrieve the component:
```

File: packages/registry/README.md [hunk 42]
```diff
@@ -0,0 +1,190 @@
...
+
+```js
+config.getComponent({
+    name: 'Teaser',
+    dependencies: ['News Item'],
+  }).component
+```
+
+The idea is that you can have both with and without dependencies:
```
Comment (stevepiercy) [inline packages/registry/README.md:168]:
```suggestion
You can have both, either with or without dependencies:
```

File: packages/registry/README.md [hunk 43]
```diff
@@ -0,0 +1,190 @@
...
+  });
+
+config.registerComponent({
+    name: 'Teaser',
+    component: MyTeaserNewsItemComponent,
+    dependencies: 'News Item',
+  });
+```
+
+and then retrieve them both, depending on the use case (in the example, given a content type value comming from `content` prop):
```
Comment (stevepiercy) [inline packages/registry/README.md:186]:
```suggestion
Then retrieve them both, depending on the use case.
In the example, given a content type value coming from the `content` prop, you would retrieve them as shown:
```

**Review by stevepiercy — CHANGES_REQUESTED** [review]
Mostly English grammar and syntax, a few one sentence per line style things, and a few questions.

File: packages/registry/README.md (hunk 23 above)
Comment (sneridagh) [reply packages/registry/README.md:57]: Not really, let me rephrase it.

File: packages/registry/package.json [hunk 44]
```diff
@@ -0,0 +1,60 @@
...
+  "name": "@plone/registry",
+  "description": "Add-on and configuration registry for Plone.",
+  "maintainers": [
+    {
+      "name": "Plone Foundation",
+      "email": "plone-developers@lists.sourceforge.net",
+      "url": "http://plone.org"
+    }
+  ],
+  "license": "MIT",
```
[the remark this answers is not in the document]
Comment (sneridagh) [reply packages/registry/package.json:11]: For Volto and related packages we decided on MIT (because it was better suited in combination with React technologies).

File: packages/registry/package.json [hunk 45]
```diff
@@ -0,0 +1,60 @@
+{
+  "name": "@plone/registry",
+  "description": "Add-on and configuration registry for Plone.",
+  "maintainers": [
+    {
+      "name": "Plone Foundation",
+      "email": "plone-developers@lists.sourceforge.net",
```
[the remark this answers is not in the document]
Comment (stevepiercy) [reply packages/registry/package.json]: Added to the agenda: https://docs.google.com/document/d/1Iafw7MuTnG3l1j3fB1TbMgtADlhJGclSk7S8K0tTXeA/edit. If possible, can a merge be delayed until after the next meeting on Thu, Nov 2?

File: packages/registry/package.json (hunk 44 above)
[the remark this answers is not in the document]
Comment (stevepiercy) [reply packages/registry/package.json:11]: Is that a problem, considering https://plone.org/foundation/copyright-licensing-logo/license-faq ? I have no idea about licensing. If we use MIT, then that page should reflect the decision to use MIT. And not just for Volto, but the cookiecutter as well.

File: packages/registry/README.md [hunk 46]
```diff
@@ -0,0 +1,206 @@
...
+    title: 'FAQ Viewer',
+    edit: FAQBlockEdit,
+    view: FAQBlockView,
+    icon: chartIcon,
+    group: 'common',
+    restricted: false,
+    mostUsed: true,
+    sidebarTab: 1,
+  };
+  ```
```
Comment (stevepiercy) [inline packages/registry/README.md:80]: Dedent the closing code fence. Also the entire code block does not need to be indented.

````suggestion
```
````

File: packages/registry/README.md [hunk 47]
```diff
@@ -0,0 +1,206 @@
...
+    restricted: false,
+    mostUsed: true,
+    sidebarTab: 1,
+  };
+
+  return config;
+}
+```
+
+Once the app starts, the add-on registry will execute, in order, all the registered add-ons' default export functions, configuring the new block, and the add-on available to the CMS when it asks the configuration registry for it.
```
Comment (stevepiercy) [inline packages/registry/README.md:106]: Hard to understand, so I broke it into two sentences.

```suggestion
Once the app starts, the add-on registry will execute, in order, all the registered add-ons' default export functions, configuring the new block.
The add-on will then become available to the CMS when it asks the configuration registry for it.
```

**Review by stevepiercy — CHANGES_REQUESTED** [review]
A couple more fixes, and it should be good to go. I also pushed up a few more words to ignore when running Vale for spell check.

File: packages/registry/package.json (hunk 44 above)
[the remark this answers is not in the document]
Comment (sneridagh) [reply packages/registry/package.json:11]: I'm not also a license expert, @polyester was looking into that, and we are ok with the Volto related packages to be MIT, in general. Maybe he can shed more light on this. Maybe that doc should be updated with it?

[the remark this answers is not in the document]
Comment (stevepiercy) [reply packages/registry/package.json:11]: OK, then I'll create an issue in plone.org to update the Licensing FAQ. Thanks for clarifying, @polyester!

[the remark this answers is not in the document]
Comment (stevepiercy) [reply packages/registry/package.json:11]: Issue created at https://github.com/plone/plone.org/issues/148. Resolving this suggestion to leave it as is.

**Comment by sneridagh** [comment]
> “There’s no point in acting surprised about it. All the planning charts and demolition orders have been on display at your local planning department in Alpha Centauri for 50 of your Earth years, so you’ve had plenty of time to lodge any formal complaint and it’s far too late to start making a fuss about it now. … What do you mean you’ve never been to Alpha Centauri? Oh, for heaven’s sake, mankind, it’s only four light years away, you know. I’m sorry, but if you can’t be bothered to take an interest in local affairs, that’s your own lookout. Energize the demolition beams.”

― Douglas Adams, [The Hitchhiker's Guide to the Galaxy](https://www.goodreads.com/work/quotes/3078186)

Quoting master Adams, as nobody filled any complain :) I'm about to merge this one, starting the 18 alpha phase, continuing with our plans for world domination.

/cc @plone/volto-team

File: packages/registry/package.json (hunk 45 above)
[the remark this answers is not in the document]
Comment (stevepiercy) [reply packages/registry/package.json]: I visited Alpha Centauri to the planning department, rappelled to the cellar with a headlamp, into the disused lavatory, pried opened the stuck file cabinet, fought off the leopard to view the display, then filed a complaint.

This one was overlooked, but it should not block merging. Instead I can open a new issue until we can bring it to the steering committee or we can make an executive decision for what should be the effective email address, if any.
